contribs/gmf/src/directives/layertree.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * @fileoverview This file provides the "layertree" directive for GeoMapFish
+ * applications.
+ *
+ * FIXME
+ */
+goog.provide('gmf.layertreeDirective');
+
+goog.require('gmf');
+goog.require('ngeo.LayertreenodeController');
+goog.require('ngeo.layertreeDirective');
+goog.require('ol.layer.Image');
+goog.require('ol.source.ImageWMS');
+
+
+/**
+ * @return {angular.Directive} The directive Definition Object.
+ * @ngInject
+ */
+gmf.layertreeDirective = function() {
+  return {
+    scope: {
+      'getMapFn': '&gmfLayertreeMap',
```

Comment:
> I could have asked the question in your previous PR: does "getMapFn" stand for "getMap function"?

Yes, it's a function that evaluates the expression passed to `gmf-layertree-map` in the context of the **parent** scope.

> If yes, what do you mean since in the HTML code "gmf-layertree-map" is a map object?

In the HTML code we have `gmf-layertree-map="mainCtrl.map"`. `"mainCtrl.map"` is an expression, which when evaluated returns the `ol.Map` instance. When we use `getMapFn: '&gmfLayertreeMap'` (note the `&` here) we tell Angular to create a function (`getMapFn`) that evaluates the expression in the context of the parent/user scope.